src/vuestic-theme/vuestic-components/va-button/VaButton.spec.js

```diff
@@ -0,0 +1,193 @@
+
+describe('VaButton', () => {
+  // 1. Button types
+
+  // True
+  it('default button', () => {
+    const wrapper = shallowMount(VaButton)
+
+    expect(wrapper.props('tag')).toBe('button')
+  })
+  // True
+  it('outline button', () => {
+    const wrapper = shallowMount(VaButton, {
+      propsData: {
+        outline: true
+      }
+    })
+    expect(wrapper.props('outline')).toBe(true)
+  })
+  // True
+  it('flat button', () => {
+    const wrapper = shallowMount(VaButton, {
+      propsData: {
+        flat: true
+      }
+    })
+    expect(wrapper.props('flat')).toBe(true)
+  })
+  // False, this type of button doesn't exist
+  it('rounded button', () => {
+    const wrapper = shallowMount(VaButton, {
+      propsData: {
+        rounded: true
+      }
+    })
+    expect(wrapper.props('rounded')).toBe(true)
+  })
+
+  // 2. Button sizes
+
+  // True
+  it('large button', () => {
+    const wrapper = shallowMount(VaButton, {
+      propsData: { large: true }
+    })
+    expect(wrapper.props('large')).toBe(true)
+  })
+  // True
+  it('small button', () => {
+    const wrapper = shallowMount(VaButton, {
+      propsData: { small: true }
+    })
+    expect(wrapper.props('small')).toBe(true)
+  })
+  // False, this button size doesn't supported
+  it('micro button', () => {
+    const wrapper = shallowMount(VaButton, {
+      propsData: { micro: true }
+    })
+    expect(wrapper.props('micro')).toBe(true)
+  })
+
+  // 3. Button colors
+
+  // True
+  it('success button', () => {
+    let cmp = shallowMount(VaButton)
+
+    expect(cmp.find(VaButton).classes()).toContain('va-button--success')
+  })
+  // True
+  it('info button', () => {
+    let cmp = shallowMount(VaButton, {
+      propsData: { color: 'info' }
+    })
+    expect(cmp.find(VaButton).classes()).toContain('va-button--info')
+  })
+  // True
+  it('danger button', () => {
+    let cmp = shallowMount(VaButton, {
+      propsData: { color: 'danger' }
+    })
+
+    expect(cmp.find(VaButton).classes()).toContain('va-button--danger')
+  })
+  // True
+  it('warning button', () => {
+    let cmp = shallowMount(VaButton, {
+      propsData: { color: 'warning' }
+    })
+
+    expect(cmp.find(VaButton).classes()).toContain('va-button--warning')
+  })
+  // True
+  it('gray button', () => {
+    let cmp = shallowMount(VaButton, {
+      propsData: { color: 'gray' }
+    })
+
+    expect(cmp.find(VaButton).classes()).toContain('va-button--gray')
+  })
+  // True
+  it('dark button', () => {
+    let cmp = shallowMount(VaButton, {
+      propsData: { color: 'dark' }
+    })
+
+    expect(cmp.find(VaButton).classes()).toContain('va-button--dark')
+  })
+  // False, color property doesn't support this value
+  it('purple button', () => {
+    let cmp = shallowMount(VaButton, {
+      propsData: { color: 'purple' }
+    })
+
+    expect(cmp.find(VaButton).classes()).toContain('va-button--purple')
+  })
+
+  // 4. Button tags
+
+  // True
+  it('a button', () => {
```

Comment:
Try more diverse checks. Include case with target you mentioned before.
This is the most complex logic in button, and, naturally, it should be tested more.